sw/host/opentitanlib/src/util/usb.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{ensure, Result};
+use rusb;
+use std::time::Duration;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Could not find USB device")]
+    NotFound,
+    #[error("Found multiple USB devices, use --serial")]
+    MultipleDevices,
+}
+
+/// The `UsbBackend` provides low-level USB access to debugging devices.
+pub struct UsbBackend {
+    device: rusb::Device<rusb::GlobalContext>,
+    handle: rusb::DeviceHandle<rusb::GlobalContext>,
+    serial_number: String,
+    timeout: Duration,
+}
+
+impl UsbBackend {
```

Comment:
Most of these methods are missing comments.

Reply:
My intention was that my multiline comments would cover 5 and 4 methods respectively with one comment.  Particularly the five "ugly" method that leak rusb data types, and should probably be changed in the future.  I do not want to spend too much time documenting the current behavior on each of them.

Reply:
Could you document your intention, then? It is very useful to try to make the output of `cargo doc` maximally friendly.

Reply:
I have now documented the four "nice" methods, while leaving the five enumeration-related methods with just the one existing block comment, stating that the API they provide is ugly in that it leaks details of the rusb implementation.  In order to refine this into a proper trait, we need to find a better interface.